platforms/ide/problems/build.gradle.kts

```diff
@@ -16,6 +16,7 @@
 
 plugins {
     id("gradlebuild.distribution.api-java")
+    id("gradlebuild.distribution.implementation-kotlin")
```

Comment:
I double checked and all modules that apply this `implementation-kotlin` plugin have sources in `main` exclusively in Kotlin. Given that this is just for one class and the potential complications with the ArchUnit test, I would suggest rewriting this one class in Java.

Reply:
I'll move the class to a different package.

Reimplementing it has also more complications than just this class. e.g. extension methods.

Reply:
I guess that could work, and we can merge it like this. Though, I am still a bit skeptical about mixing Kotlin with Java. It should work, but I don't know if there was a reason why we never did it.

The ArchUnit is also something we need to improve, because it becomes too easy to include packages that contain both Kotlin and Java.